redis_commands.c

```diff
@@ -2047,26 +2047,51 @@ int redis_pfcount_cmd(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock,
     return SUCCESS;
 }
 
+char *redis_variadic_str_cmd(char *kw, zval *argv, int argc, int *cmd_len) {
+    smart_string cmdstr = {0};
+    zend_string *zstr;
+    int i;
+
+    redis_cmd_init_sstr(&cmdstr, argc, kw, strlen(kw));
+
+    for (i = 0; i < argc; i++) {
+        zstr = zval_get_string(&argv[i]);
+        redis_cmd_append_sstr_zstr(&cmdstr, zstr);
+        zend_string_release(zstr);
+    }
+
+    *cmd_len = cmdstr.len;
+    return cmdstr.c;
+}
+
 int redis_auth_cmd(INTERNAL_FUNCTION_PARAMETERS, RedisSock *redis_sock,
                    char **cmd, int *cmd_len, short *slot, void **ctx)
 {
-    char *pw;
-    size_t pw_len;
+    zend_string *user, *pass;
+    zval *ztest;
 
-    if (zend_parse_parameters(ZEND_NUM_ARGS(), "s", &pw, &pw_len)
-                             ==FAILURE)
+    if (zend_parse_parameters(ZEND_NUM_ARGS(), "z!", &ztest) == FAILURE ||
+        redis_extract_auth_info(ztest, &user, &pass) == FAILURE)
     {
         return FAILURE;
     }
 
-    // Construct our AUTH command
-    *cmd_len = REDIS_CMD_SPPRINTF(cmd, "AUTH", "s", pw, pw_len);
+    /* We weren't provided user or password, short circuit */
+    if (!user && !pass)
```

Comment:
should this be `||` instead of `&&`?

Reply:
No, it's ok if they only send `pass`.

That would have us `AUTH <pass>` (backward compatible)
When we have `user` and `pass` it will send `AUTH <user> <pass>` (new Redis 6 feature)

That said it might be better to just add a third `else` and return FAILURE so the logic is easier to follow.